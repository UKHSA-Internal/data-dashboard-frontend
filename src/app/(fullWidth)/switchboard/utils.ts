import assert from 'assert'
import { cookies } from 'next/headers'

import { UKHSA_SWITCHBOARD_COOKIE_NAME } from '@/app/constants/app.constants'

import { switchBoardSchema } from './schema'
import { getSwitchBoardState, initialState } from './state'

const apiSchema = switchBoardSchema.shape.api

export type FieldId = keyof (typeof initialState)['api']

function pickFieldState(fieldId: FieldId, form: FormData) {
  assert(fieldId)
  assert(form)

  const selected = form.get(`${fieldId}.selected`)
  const status = form.get(`${fieldId}.status`)

  return apiSchema.shape[fieldId].parse({
    selected: selected !== '' ? selected : null,
    status,
  })
}

export function mergeFieldState(fieldId: FieldId, form: FormData) {
  assert(fieldId)
  assert(form)

  const cookieStore = cookies()
  const fieldState = pickFieldState(fieldId, form)
  const switchBoardState = getSwitchBoardState(cookieStore.get(UKHSA_SWITCHBOARD_COOKIE_NAME)?.value)
  const newState: switchBoardSchema['api'] = {
    [fieldId]: fieldState,
  }

  cookieStore.set(
    UKHSA_SWITCHBOARD_COOKIE_NAME,
    JSON.stringify({
      ...switchBoardState,
      api: {
        ...switchBoardState.api,
        ...newState,
      },
    })
  )
}

export const handleSubmitAction = async (form: FormData) => {
  'use server'
  const id = form.get('id') as FieldId
  assert(id)
  mergeFieldState(id, form)
}
