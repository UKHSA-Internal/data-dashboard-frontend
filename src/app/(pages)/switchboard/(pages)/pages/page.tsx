import { cookies } from 'next/headers'

import { StatusSelect } from '@/app/(pages)/switchboard/components/StatusSelect'
import { heading } from '@/app/(pages)/switchboard/shared/constants'
import { getSwitchBoardState, syncState } from '@/app/(pages)/switchboard/shared/state'
import { Details } from '@/app/components/ui/govuk'
import { View } from '@/app/components/ui/ukhsa'
import { UKHSA_SWITCHBOARD_COOKIE_NAME } from '@/app/constants/app.constants'

export default function SwitchBoard() {
  const cookieStore = cookies()

  const {
    api: { pages },
  } = getSwitchBoardState(cookieStore.get(UKHSA_SWITCHBOARD_COOKIE_NAME)?.value)

  return (
    <View heading={heading} className="govuk-!-margin-top-5" backLink="/switchboard">
      <form
        className="govuk-!-margin-top-3"
        action={async (form) => {
          'use server'

          syncState({
            pages: {
              list: {
                status: Number(form.get('pages.list.status')),
              },
              detail: {
                status: Number(form.get('pages.detail.status')),
                scenario: {
                  relatedLinksLayout: form.get('pages.detail.scenario.relatedLinksLayout'),
                },
              },
            },
          })
        }}
      >
        <fieldset className="govuk-fieldset govuk-!-margin-bottom-6">
          <legend className="govuk-fieldset__legend govuk-fieldset__legend--m">
            <h2 className="govuk-fieldset__heading">Pages</h2>
          </legend>
          <StatusSelect id="pages.list.status" name="pages.list.status" defaultValue={pages.list.status} />
        </fieldset>

        <fieldset className="govuk-fieldset govuk-!-margin-bottom-6">
          <legend className="govuk-fieldset__legend govuk-fieldset__legend--m">
            <h2 className="govuk-fieldset__heading">Page</h2>
            <span className="govuk-hint">Overides all individual page requests</span>
          </legend>
          <StatusSelect id="pages.detail.status" name="pages.detail.status" defaultValue={pages.detail.status} />

          <Details label="Related Links" className="govuk-!-margin-top-5">
            <div className="govuk-radios govuk-radios--small" data-module="govuk-radios">
              <div className="govuk-radios__item">
                <input
                  defaultChecked={pages.detail.scenario.relatedLinksLayout === 'Sidebar'}
                  className="govuk-radios__input"
                  id="pages.detail.scenario.relatedLinksLayout.Sidebar"
                  name="pages.detail.scenario.relatedLinksLayout"
                  type="radio"
                  value="Sidebar"
                />
                <label
                  className="govuk-label govuk-radios__label"
                  htmlFor="pages.detail.scenario.relatedLinksLayout.Sidebar"
                >
                  Sidebar
                </label>
              </div>

              <div className="govuk-radios__item">
                <input
                  defaultChecked={pages.detail.scenario.relatedLinksLayout === 'Footer'}
                  className="govuk-radios__input"
                  id="pages.detail.scenario.relatedLinksLayout.Footer"
                  name="pages.detail.scenario.relatedLinksLayout"
                  type="radio"
                  value="Footer"
                />
                <label
                  className="govuk-label govuk-radios__label"
                  htmlFor="pages.detail.scenario.relatedLinksLayout.Footer"
                >
                  Footer
                </label>
              </div>
            </div>
          </Details>
        </fieldset>

        <button type="submit" className="govuk-button">
          Save changes
        </button>
      </form>
    </View>
  )
}
