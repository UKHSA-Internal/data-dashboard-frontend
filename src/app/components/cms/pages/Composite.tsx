import { PageType } from '@/api/requests/cms/getPages'
import { getPageBySlug } from '@/api/requests/getPageBySlug'
import { View } from '@/app/components/ui/ukhsa'
import { PageComponentBaseProps } from '@/app/types'
import { renderCompositeBlock } from '@/app/utils/cms.utils'

import { RelatedLinksWrapper } from '../../ui/ukhsa/RelatedLinks/RelatedLinksWrapper'
import { Heading } from '../../ui/ukhsa/View/Heading/Heading'
import { LastUpdated } from '../../ui/ukhsa/View/LastUpdated/LastUpdated'
import { RichTextAutoHeadings } from '../RichText/RichTextAutoHeadings'

export default async function CompositePage({ slug }: PageComponentBaseProps) {
  const {
    title,
    body,
    last_updated_at: lastUpdated,
    related_links: relatedLinks,
    related_links_layout: relatedLinksLayout,
  } = await getPageBySlug<PageType.Common | PageType.Composite>(slug)

  return (
    <View>
      <Heading heading={title} />
      <LastUpdated lastUpdated={lastUpdated} />
      <div className="govuk-grid-row">
        <div className="govuk-grid-column-three-quarters-from-desktop">
          {typeof body === 'string' ? (
            <RichTextAutoHeadings>{body}</RichTextAutoHeadings>
          ) : (
            body.map(renderCompositeBlock)
          )}
        </div>

        {relatedLinksLayout === 'Sidebar' ? (
          <div className="govuk-grid-column-one-quarter-from-desktop govuk-!-margin-top-2 sticky top-2">
            <RelatedLinksWrapper layout={relatedLinksLayout} links={relatedLinks} />
          </div>
        ) : null}
      </div>

      {relatedLinksLayout === 'Footer' ? (
        <RelatedLinksWrapper layout={relatedLinksLayout} links={relatedLinks} />
      ) : null}
    </View>
  )
}
