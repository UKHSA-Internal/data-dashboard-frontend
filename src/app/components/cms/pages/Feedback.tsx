import { getFeedbackPage } from '@/app/utils/cms'

import { View } from '../../ui/ukhsa'
import { Heading } from '../../ui/ukhsa/Heading/Heading'
import Feedback from '../Feedback/Feedback'

export default async function FeedbackPage() {
  const { title, form_fields: formFields } = await getFeedbackPage()

  return (
    <View>
      <Heading heading={title} />
      <Feedback formFields={formFields} />
    </View>
  )
}
