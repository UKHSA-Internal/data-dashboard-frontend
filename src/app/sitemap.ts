import dayjs from 'dayjs'
import { MetadataRoute } from 'next'

import { getPages } from '@/api/requests/cms/getPages'
import { getHealthAlerts } from '@/api/requests/health-alerts/getHealthAlerts'
import { logger } from '@/lib/logger'

import { SITE_URL } from './constants/app.constants'
import { toSlug } from './utils/app.utils'

const rootUrl = `https://${SITE_URL}`

type ChangeFrequency = MetadataRoute.Sitemap[number]['changeFrequency']

export default async function sitemap(): Promise<MetadataRoute.Sitemap> {
  const pages = await getPages()

  if (pages.error) {
    logger.error('Failed to generate sitemap.xml')
    logger.error(pages.error)
  }

  const sitemap: MetadataRoute.Sitemap = []

  if (pages.success) {
    for (const page of pages.data.items) {
      const url = page.meta.html_url ?? ''

      const {
        meta: { seo_change_frequency: changeFrequency, seo_priority: priority },
      } = page

      sitemap.push({
        url,
        changeFrequency: changeFrequency.toLowerCase() as ChangeFrequency,
        priority,
        lastModified: new Date(),
      })
    }
  }

  // Non-CMS pages. TODO: Migrate these to be CMS delivered
  // Feedback
  sitemap.push({
    url: `${rootUrl}/feedback`,
    lastModified: new Date(),
    changeFrequency: 'monthly',
    priority: 0.7,
  })

  // Brwose
  sitemap.push({
    url: `${rootUrl}/browse`,
    lastModified: new Date(),
    changeFrequency: 'monthly',
    priority: 0.2,
  })

  // Weather Health Alert Regions
  const [weatherHealthAlertHeatPages, weatherHealthAlertColdPages] = await Promise.all([
    getHealthAlerts('heat'),
    getHealthAlerts('cold'),
  ])

  if (weatherHealthAlertHeatPages.success && weatherHealthAlertColdPages.success) {
    for (const page of weatherHealthAlertHeatPages.data) {
      sitemap.push({
        url: `${rootUrl}/weather-health-alerts/heat/${toSlug(page.geography_name)}`,
        lastModified: dayjs(page.refresh_date).toDate(),
        changeFrequency: 'monthly',
        priority: 0.8,
      })
    }

    for (const page of weatherHealthAlertHeatPages.data) {
      sitemap.push({
        url: `${rootUrl}/weather-health-alerts/cold/${toSlug(page.geography_name)}`,
        lastModified: dayjs(page.refresh_date).toDate(),
        changeFrequency: 'monthly',
        priority: 0.8,
      })
    }
  }

  return sitemap
}
