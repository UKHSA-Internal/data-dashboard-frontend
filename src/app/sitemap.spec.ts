import { client } from '@/api/utils/api.utils'
import { allPagesMock } from '@/mock-server/handlers/cms/pages/fixtures/pages'

import sitemap from './sitemap'

const getPages = jest.mocked(client)

test('Sitemap displays correctly', async () => {
  // Mock date as some sitemap pages have their last modified date generated at build time
  jest.useFakeTimers().setSystemTime(new Date('2020-01-01'))

  getPages.mockResolvedValueOnce({
    status: 200,
    data: allPagesMock,
  })

  expect(await sitemap()).toMatchSnapshot()
})
