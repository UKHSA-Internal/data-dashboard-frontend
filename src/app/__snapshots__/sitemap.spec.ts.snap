// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sitemap displays correctly 1`] = `
[
  {
    "changeFrequency": "weekly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 1,
    "url": "http://localhost:3000/dashboard",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.7,
    "url": "http://localhost:3000/about",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.5,
    "url": "http://localhost:3000/cookies",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.5,
    "url": "http://localhost:3000/accessibility-statement",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.5,
    "url": "http://localhost:3000/compliance",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.5,
    "url": "http://localhost:3000/archive-data",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.7,
    "url": "http://localhost:3000/bulk-downloads",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.5,
    "url": "http://localhost:3000/access-our-data",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.5,
    "url": "http://localhost:3000/access-our-data/overview",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.5,
    "url": "http://localhost:3000/access-our-data/what-is-an-api",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.5,
    "url": "http://localhost:3000/access-our-data/getting-started",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.5,
    "url": "http://localhost:3000/access-our-data/api-authentication",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.5,
    "url": "http://localhost:3000/access-our-data/data-structure",
  },
  {
    "changeFrequency": "yearly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.8,
    "url": "http://localhost:3000/weather-health-alerts",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.9,
    "url": "http://localhost:3000/weather-health-alerts/cold",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.9,
    "url": "http://localhost:3000/weather-health-alerts/heat",
  },
  {
    "changeFrequency": "weekly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.8,
    "url": "http://localhost:3000/dashboard/covid-19",
  },
  {
    "changeFrequency": "weekly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.8,
    "url": "http://localhost:3000/dashboard/influenza",
  },
  {
    "changeFrequency": "weekly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.8,
    "url": "http://localhost:3000/dashboard/other-respiratory-viruses",
  },
  {
    "changeFrequency": "weekly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.7,
    "url": "http://localhost:3000/whats-new",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/soft-launch-of-the-ukhsa-data-dashboard",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/updated-csv-download-and-tabular-data-functionality",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/other-respiratory-viruses-data-added-to-the-homepage",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-1",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-2",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-3",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-4",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-5",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-6",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-7",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-8",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-9",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-10",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-11",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-12",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-13",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-14",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-15",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-16",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-17",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-18",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-19",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-20",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-21",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-22",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-23",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-24",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-25",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-26",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-27",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-28",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-29",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/whats-new/whats-new-child-mock-30",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.5,
    "url": "http://localhost:3000/metrics-documentation",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/new-cases-7days-sum",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/new-deaths-7days-change-percentage",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/covid-occupied-beds-latest",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/new-pcr-tests-daily",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/new-cases-7days-percent",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-1",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-2",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-3",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-4",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-5",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-6",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-7",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-8",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-9",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-10",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-11",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-12",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-13",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-14",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-15",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-16",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-17",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-18",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-19",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-20",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-21",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-22",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-23",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-24",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-25",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-26",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-27",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-28",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-29",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-30",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-31",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-32",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-33",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-34",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-35",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-36",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-37",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-38",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-39",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-40",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-41",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-42",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-43",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-44",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-45",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-46",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-47",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-48",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-49",
  },
  {
    "changeFrequency": "never",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.4,
    "url": "http://localhost:3000/metrics-documentation/metrics-child-mock-50",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.7,
    "url": "https://ukhsa-dashboard.data.gov.uk/feedback",
  },
  {
    "changeFrequency": "monthly",
    "lastModified": 2020-01-01T00:00:00.000Z,
    "priority": 0.2,
    "url": "https://ukhsa-dashboard.data.gov.uk/browse",
  },
]
`;
